src/scene/materials/standard-material.js

```diff
@@ -920,9 +924,9 @@ const defineProp = (prop) => {
     return prop.defaultValue && prop.defaultValue.clone ? defineAggProp(prop) : defineValueProp(prop);
 };
 
-function _defineTex2D(name, uv, channels, defChannel, vertexColor, detailMode) {
+function _defineTex2D(name, channel = "rgb", vertexColor = true, uv = 0) {
     // store texture name
-    _matTex2D[name] = channels;
+    _matTex2D[name] = channel.length || -1;
```

Comment:
but strange to store -1 as texture name?

Reply:
Yup agree (this is not new).